Fixes unequal length bug present in Anagram class

annreillyzimmerman opened the pull request:
Checks that both Strings fed into Anagram class are of equal length

annreillyzimmerman committed:
Fixes unequal length bug present in Anagram class

Checks that both Strings fed into Anagram class are of equal length

peterstory reviewed (changes requested):
Although this may allow the test cases to pass, this fix actually doesn't entirely fix the bug. Also, you should avoid committing changes to IDE configuration files (e.g., things in the .idea directory).

annreillyzimmerman committed:
Bugfix for repeating characters in anagram

Replaces <Character, Boolean> hashmap with <Character, Integer> in order to count number of characters in anagram

peterstory reviewed (changes requested):
I think you have the right idea, but your solution should still be more general. 

Instead of setting to 1 or 2, why not set to 1 or increment? 

Also, please revert your change to the .idea file.

annreillyzimmerman committed:
Fixes previous bug
